### PR TITLE
i/b/desktop: allow access to fcitx portal which is supported by both fcitx/fcitx5

### PR DESCRIPTION
**Pull request by hantengc**
References:
https://github.com/fcitx/fcitx/blob/master/src/frontend/ipcportal/ipcportal.h#L27 https://github.com/fcitx/fcitx5/blob/master/src/frontend/dbusfrontend/dbusfrontend.cpp#L484

### CODE REVIEW

**Comment by alexmurray** [comment]
I notice that the `desktop-legacy` interface already provides some access to `fcitx` - can you please check if this is sufficient?

**Comment by hantengc** [comment]
> I notice that the `desktop-legacy` interface already provides some access to `fcitx` - can you please check if this is sufficient?

Not a thing.

**Comment by alexmurray** [comment]
> Not a thing.

I'm sorry but I don't understand what you mean by this comment - can you please provide more information?

**Comment by wengxt** [comment]
Hi, fcitx dev here.

So some background about fcitx's dbus interface:
fcitx4 uses a separate dbus-daemon and uses a separate dbus file, the main reason for that is because in the past we frequently got report due to people just didn't launch the session bus properly, that's what defined in desktop_legacy.

When flatpak becomes a thing, we decide instead of using a separate dbus-daemon, we will use this opportunity to create a new service name accepted by xdg-dbus-proxy and keep using it for fcitx5. This new dbus interface was supported since 4.2.9.5+, which is released in 2017.

So this is the interface being actively maintained in fcitx5 and enables new features that works on wayland better.

**Comment by alexmurray** [comment]
Thanks for the excellent background information @wengxt - that definitely helps to clarify the purpose of this new PR.

File: interfaces/builtin/desktop.go [hunk 1]
```diff
@@ -308,6 +308,20 @@ dbus (send, receive)
...
+
+# Allow access to the Fcitx portal, supported by fcitx/fcitx5
+dbus (send)
+      bus=session
+      path=/{,org/freedesktop/portal/}inputmethod
+      interface=org.fcitx.Fcitx.InputMethod1
+      member={CreateInputContext,Version}
+      peer=(name=org.freedesktop.portal.Fcitx),
+
+      dbus (send, receive)
```
Comment (alexmurray) [inline interfaces/builtin/desktop.go:320]: Can you please fix the indentation of this line - `dbus (send, receive)` should not be indented.

**Review by alexmurray — CHANGES_REQUESTED** [review]
LGTM but can you please fix the indentation of the second rule?

File: interfaces/builtin/desktop.go (hunk 1 above)
Comment (hantengc) [reply interfaces/builtin/desktop.go]: ok ,It's done. @alexmurray

**Comment by hantengc** [comment]
@alexmurray    hi ~ I have signed the CLA, how do I trigger the redetection mechanism?

**Comment by alexmurray** [comment]
@ hantengc the easiest way is to close and reopen the PR which will retrigger all the automated checks - otherwise I think @mvo5 can probably retrigger just that one check if you want to wait until he is online.

**Comment by hantengc** [comment]
@mvo5   hi~ take a look at the above test for cla-check

**Review by mvo5 — APPROVED** [review]
Thank you!

**Comment by hantengc** [comment]
@alexmurray  What is the specific reason for the above error, can you help me explain? I didn't understand why there was a problem with the ubuntu installation package

**Review by alexmurray — APPROVED** [review]
LGTM!

**Comment by alexmurray** [comment]
> @alexmurray What is the specific reason for the above error, can you help me explain? I didn't understand why there was a problem with the ubuntu installation package

This looks like a transient network issue - I suggest closing and then immediately reopening this PR to retrigger all the automated tests to run.

**Comment by hantengc** [comment]
@mvo5 @alexmurray  cla check

**Comment by hantengc** [comment]
@alexmurray @mvo5 what's the problem???

**Comment by alexmurray** [comment]
The test failures appear unrelated to this change from what I can see, I don't think there is anything you need to do @hantengc

**Comment by hantengc** [comment]
> The test failures appear unrelated to this change from what I can see, I don't think there is anything you need to do @hantengc

Combining one PR is too much trouble.

**Pull request merged**